dcx_flight_env_expansion copy.py

```diff
@@ -214,31 +253,54 @@ def compute_heading_error(current_heading, target_heading):
 pitch_lpf = LPF(4,1,10.0)
 prev_dist = 0
 vel_sign = -1
+expected_frame_time = (1/dcx.CLOCK_RATE) * 1e9
+def dynamic_sleep(actual_frame_time: float):
```

Comment:
should pull this into a util svc since it can be reused in most cyclic real time avionics for more determinism